pkg/services/playlist/playlistimpl/sqlx_store.go

```diff
@@ -0,0 +1,248 @@
+package playlistimpl
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/playlist"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/jmoiron/sqlx"
+)
+
+type sqlxStore struct {
+	sqlxdb  *sqlx.DB
+	dialect migrator.Dialect
+}
+
+func (s *sqlxStore) WithTransaction(ctx context.Context, db *sqlx.DB, fn func(*sqlx.Tx) error) error {
+	// In the same transaction is quaranteed in the same session
+	tx, err := db.Beginx()
+	if err != nil {
+		return err
+	}
+	// Here when I try to set the session sql_mode, other integration test is failed
+	if s.dialect.DriverName() == "mysql" {
+		_, err = tx.ExecContext(ctx, "SET SESSION sql_mode = 'ANSI_QUOTES'")
```

Comment:
here we make sure that in the session we use double quote, since we can make sure all in one transaction should belong to the same session? Checked in standard library, seems golang doesn't support db session?

Reply:
this wrapper should go to db.DB ideally